src/main/java/games/strategy/engine/data/TripleaHandler.java

```diff
@@ -0,0 +1,434 @@
+    requiredParents.put(XML_NODE_NAME_UNIT_PLACEMENT, XML_NODE_NAME_UNIT_INITIALIZE);
+    requiredParents.put(XML_NODE_NAME_RESOURCE_GIVEN, XML_NODE_NAME_RESOURCE_INITIALIZE);
+    requiredParents.put(XML_NODE_NAME_PROPERTY, XML_NODE_NAME_PROPERTY_LIST);
+    requiredParents.put(XML_NODE_NAME_NUMBER, XML_NODE_NAME_PROPERTY);
+    requiredParents.put(XML_NODE_NAME_BOOLEAN, XML_NODE_NAME_PROPERTY);
+    requiredParents.put(XML_NODE_NAME_VALUE, XML_NODE_NAME_PROPERTY);
+    requiredParents.put(XML_NODE_NAME_STRING, XML_NODE_NAME_PROPERTY);
+    requiredParents.put(XML_NODE_NAME_TECHNOLOGIES, XML_NODE_NAME_TECHNOLOGY);
+    requiredParents.put(XML_NODE_NAME_TECHNAME, XML_NODE_NAME_TECHNOLOGIES);
+    requiredParents.put(XML_NODE_NAME_PLAYER_TECH, XML_NODE_NAME_TECHNOLOGY);
+    requiredParents.put(XML_NODE_NAME_CATEGORY, XML_NODE_NAME_PLAYER_TECH);
+    requiredParents.put(XML_NODE_NAME_TECH, XML_NODE_NAME_CATEGORY);
+    requiredParents.put(XML_NODE_NAME_REPAIR_RULES, XML_NODE_NAME_REPAIR_FRONTIER);
+    requiredParents.put(XML_NODE_NAME_RELATIONSHIP_TYPE, XML_NODE_NAME_RELATIONSHIP_TYPES);
+    requiredParents.put(XML_NODE_NAME_TERRITORY_EFFECT, XML_NODE_NAME_TERRITORY_EFFECT_LIST);
+    requiredParents.put(XML_NODE_NAME_STEP_PROPERTY, XML_NODE_NAME_STEP);
+  }
+
+  public TripleaHandler(GameData gameData) {
+    this.gameData = gameData;
+  }
+
+  @Override
+  public void startElement(String uri, String localName, String qname, Attributes attributes) throws SAXException {
+    try {
+      checkParentNode(qname);
+    } catch (GameParseException e) {
+      throw new SAXException(e);
+    }
+    handleElement(qname, attributes);
+    stack.addFirst(new PseudoElement(qname, new AttributesImpl(attributes)));
+  }
+
+  @Override
+  public void endElement(String uri, String localName, String qname) throws SAXException {
+    stack.pop();
+    try {
+      switch (qname) {
+        case XML_NODE_NAME_MAP:
+          handleMap();
+          break;
+        case XML_NODE_NAME_PLAYER_LIST:
+          handlePlayerList();
+          break;
+        case XML_NODE_NAME_PRODUCTION:
+          handleProduction();
+          break;
+        case XML_NODE_NAME_PRODUCTION_RULE:
+          handleProductionRule();
+          break;
+        case XML_NODE_NAME_PRODUCTION_FRONTIER:
+          handleProductionFrontier();
+          break;
+        case XML_NODE_NAME_GAME_PLAY:
+          handleGamePlay();
+          break;
+        case XML_NODE_NAME_ATTACHMENT_LIST:
+          handleAttachmentList();
+          break;
+        case XML_NODE_NAME_ATTACHMENT:
+          handleAttachment();
+          break;
+        case XML_NODE_NAME_INITIALIZE:
+          handleInitialize();
+          break;
+        case XML_NODE_NAME_PROPERTY_LIST:
+          handlePropertyList();
+          break;
+        default:
+          break;
+      }
+    } catch (GameParseException e) {
+      throw new SAXException(e);
+    }
```

Comment:
Now I understand why you were inclined to have `GameParseException` extend `SAXException`.  However, I think it's ok for all of the `TripleaHandler` template methods to throw `SAXException` instead of `GameParseException`.

`GameParseException` seems to be used by `GameParser` to communicate domain-specific errors not necessarily related to parsing (e.g. game data validation).  Therefore, it should probably only be thrown by the public methods of `GameParser`.

On the other hand, `TripleaHandler` is effectively part of the SAX parser.  Therefore, its domain-specific exception type is naturally `SAXException`.  Simply changing all occurrences of `GameParseException` to `SAXException` in `TripleaHandler` (and its anonymous subclass in `MapXmlHelper`) seems to work and allows you to get rid of the recently-added try-catch blocks that wrap a `GameParseException` in a `SAXException`.

One could argue that `GameParser#parse()` shouldn't even throw `SAXException`, and that it should always be wrapped in a `GameParseException`, but that's probably outside the scope of this PR.